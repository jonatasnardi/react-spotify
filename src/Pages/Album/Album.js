import React, { Component } from 'react';
import "./Album.scss";
import Iframe from 'react-iframe'
import { withRouter } from 'react-router-dom'
import {  ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import api from '../../Services/api';
import { millisToMinutesAndSeconds } from '../../Helpers/functions';

class Album extends Component {
  constructor(props) {
    super(props);

    this.state = {
      albumId: this.props.match.params.id,
      albumInfo: null,
      albumTracks: [],
      previewUrl: null,
    };

    this.requestAlbum();
    this.requestTracks();
  }

  requestAlbum = () => {
    const { albumId } = this.state;
    
    api.get(`/albums/${albumId}`, {})
      .then(response => { 
        this.setState({
          albumInfo: response.data
        });
      })
      .catch(error => {
        this.props.history.push('/login');
      });
  }

  requestTracks = () => {
    const { albumId } = this.state;

    api.get(`/albums/${albumId}/tracks`, {})
      .then(response => { 
        this.setState({
          albumTracks: response.data.items,
        });
      })
      .catch(error => {
        this.props.history.push('/login');
      });
  }

  playMusic = (previewUrl) => {
    if (previewUrl) {
      this.setState({
        previewUrl
      });
    } else {
      toast.info(`Esta música não possui preview`, {
        position: toast.POSITION.TOP_RIGHT
      });
    }
  }

  render() {
    const { albumInfo, albumTracks, previewUrl } = this.state;
    
    return (
      <section className="album">
        <ToastContainer autoClose={3000} />

        <h3 className="back-label"><a onClick={() => this.props.history.push('/home')}>‹ Voltar</a></h3>

        <div className="box-album-info">
          { albumInfo && (
            <>
              <div className="container-image">
                <img src={albumInfo.images[0].url} className="album-image" alt="album" />
              </div>

              <div className="container-text">
                <p className="album-title">{albumInfo.name}</p>
                <p className="album-artist">{albumInfo.artists[0].name}</p>
              </div>
            </>
          )}
        </div>

        <div className="box-album-tracks">
          <ul className="group-tracks">
            { albumTracks && albumTracks.map((track, i) => (
                <li key={track} className={'track-item'} onClick={() => this.playMusic(track.preview_url)}>
                  <div className="container-text">
                    <span className="track-index">{i + 1}.</span>
                    <span className="track-title">{track.name}</span>
                    <span className="track-time">{millisToMinutesAndSeconds(track.duration_ms)}</span>
                  </div>
                </li>
              ))
            }
          </ul>
        </div>

        { previewUrl && (
          <Iframe url={previewUrl}
            width="100%"
            height="200px"
            id="myId"
            className="myClassname"
            display="initial"
            position="relative"/>
        )}
        
      </section>
    );
  }
}

export default withRouter(Album);