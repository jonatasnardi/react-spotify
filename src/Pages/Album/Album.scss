@import "../../Styles/styles.scss";

.album {
  .back-label {
    font-size: 16px;
    color: $terciary;
    padding: 0 5px;
    cursor: pointer;
  }

  .box-album-info {
    float: left;
    width: 30%;

    @include maxWidth(991) {
      width: 100%;
    }
  }

  .box-album-tracks {
    float: left;
    width: 70%;
    padding: 0 15px;
    @include maxWidth(991) {
      width: 100%;
    }

    .group-tracks {
      list-style: none;

      .track-item {
        width: 100%;
        cursor: pointer;
        transition: all 0.3s;
        padding: 10px 10px 0 10px;
        &:hover {
          background: lighten($secondary, 7%);
        }

        .track-index {
          width: 10%;
          float: left;
          font-size: 16px;
          color: $gray-light;
          text-align: right;
        }

        .track-title {
          width: 70%;
          float: left;
          font-size: 16px;
          color: $terciary;
          padding-left: 15px;
        }

        .track-time {
          width: 20%;
          float: left;
          font-size: 16px;
          color: $gray-light;
          text-align: right;
        }
      }
    }
  }

  .box-search {
    margin-bottom: 50px;

    .input-search {
      font-size: 48px;
      background: transparent;
      width: 100%;
      border: none;
      border-bottom: 2px solid $gray-light;
      color: $white;
      padding: 10px 5px;
    }
  }

  .container-image {
    margin-top: 20px;
    
    .album-image {
      max-width: 100%;
      margin-bottom: 20px;
    }
  }

  .container-text {
    min-height: 30px;

    .album-title {
      font-size: 16px;
      color: $terciary;
      text-align: center;
      min-height: 34px;
      max-height: 34px;
      margin-bottom: 10px;
    }
  
    .album-artist {
      font-size: 16px;
      color: $gray-light;
      text-align: center;
      min-height: 34px;
      max-height: 34px;
    }
  }
}
