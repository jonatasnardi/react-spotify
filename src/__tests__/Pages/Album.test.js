import React from 'react';
import { shallow } from 'enzyme';
import Album from '../../Pages/Album/Album';

describe('Testing Album Component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Album />
    );

    expect(wrapper).toMatchSnapshot();
  })
})