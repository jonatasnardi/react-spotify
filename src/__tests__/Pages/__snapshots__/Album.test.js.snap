// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Album Component should render correctly 1`] = `ShallowWrapper {}`;
